services/ui-src/src/components/export/ExportedReportFieldRow.test.tsx

```diff
@@ -0,0 +1,137 @@
+import { render, screen } from "@testing-library/react";
+import { axe } from "jest-axe";
+import { ExportedReportFieldRow } from "./ExportedReportFieldRow";
+import { mockWpReportContext } from "utils/testing/setupJest";
+import { ReportContext } from "components";
+import { Table } from "@chakra-ui/react";
+
+const field = {
+  id: "test",
+  validation: "string",
+  type: "drawer",
+  props: { hint: "hint", number: 123 },
+};
+const otherTextField = {
+  id: "test-otherText",
+  validation: "string",
+  type: "dynamic",
+  props: { hint: "hint" },
+};
+const fieldWithLabel = {
+  id: "test",
+  validation: "string",
+  type: "drawer",
+  props: { hint: "hint", label: "test label" },
+};
+
+const fieldWithNoLabelorHint = {
+  id: "test",
+  validation: "string",
+  type: "drawer",
+  props: {},
+};
+
+const exportRow = (
+  <ReportContext.Provider value={mockWpReportContext}>
+    <Table>
+      <tbody>
+        <ExportedReportFieldRow formField={field} pageType="drawer" />
+      </tbody>
+    </Table>
+  </ReportContext.Provider>
+);
+
+const otherTextRow = (
+  <ReportContext.Provider value={mockWpReportContext}>
+    <Table>
+      <tbody>
+        <ExportedReportFieldRow formField={otherTextField} pageType="drawer" />
+      </tbody>
+    </Table>
+  </ReportContext.Provider>
+);
+
+const dynamicRow = (
+  <ReportContext.Provider value={mockWpReportContext}>
+    <Table>
+      <tbody>
+        <ExportedReportFieldRow formField={fieldWithLabel} pageType="drawer" />
+      </tbody>
+    </Table>
+  </ReportContext.Provider>
+);
+
+const noHintRow = (
+  <ReportContext.Provider value={mockWpReportContext}>
+    <Table>
+      <tbody>
+        <ExportedReportFieldRow
+          formField={field}
+          pageType="drawer"
+          showHintText={false}
+        />
+      </tbody>
+    </Table>
+  </ReportContext.Provider>
+);
+
+const noHintNoLabel = (
+  <ReportContext.Provider value={mockWpReportContext}>
+    <Table>
+      <tbody>
+        <ExportedReportFieldRow
+          formField={fieldWithNoLabelorHint}
+          pageType="drawer"
+        />
+      </tbody>
+    </Table>
+  </ReportContext.Provider>
+);
+
+describe("ExportedReportFieldRow", () => {
+  test("Is present", async () => {
+    render(exportRow);
+    const row = screen.getByTestId("exportRow");
+    expect(row).toBeVisible();
+  });
+
+  test("is not visible when ", async () => {
+    render(otherTextRow);
+    const row = screen.queryByTestId("exportRow");
+    expect(row).toBeNull();
```

Comment:
in cases like this I think `not.toBeInDocument()` _might_ be better? I think I remember that null is a little too forgiving, though maybe here it really doesn't make a difference!